src/main/java/org/folio/dataimport/handlers/actions/CreateInvoiceEventHandler.java

```diff
@@ -372,9 +381,45 @@ private List<InvoiceLine> prepareInvoiceLinesToSave(String invoiceId, DataImport
       .collect(Collectors.toList());
 
     linkInvoiceLinesToPoLines(invoiceLines, associatedPoLines);
+    ensureFundCode(invoiceLines, dataImportEventPayload);
     return invoiceLines;
   }
 
+  private void ensureFundCode(List<InvoiceLine> invoiceLines, DataImportEventPayload dataImportEventPayload) {
+    if (invoiceLines.stream().allMatch(line -> isEmpty(line.getFundDistributions()))) {
+      return;
+    }
+
+    Map<String, String> idToFundName = extractFundsData(dataImportEventPayload);
+    if (!idToFundName.isEmpty()) {
+      invoiceLines.stream()
+        .filter(invoiceLine -> isNotEmpty(invoiceLine.getFundDistributions()))
+        .forEach(invoiceLine -> invoiceLine.getFundDistributions().stream()
+          .filter(fundDistribution -> isNotEmpty(fundDistribution.getFundId()) && isEmpty(fundDistribution.getCode()))
+          .forEach(fundDistribution -> populateFundCode(fundDistribution, idToFundName)));
+    }
+  }
+
+  private Map<String, String> extractFundsData(DataImportEventPayload dataImportEventPayload) {
+    ProfileSnapshotWrapper mappingProfileWrapper = dataImportEventPayload.getCurrentNode();
+    MappingProfile mappingProfile = ObjectMapperTool.getMapper().convertValue(mappingProfileWrapper.getContent(), MappingProfile.class);
+
+    return mappingProfile.getMappingDetails().getMappingFields().stream()
+      .filter(mappingRule -> INVOICE_LINES_RULE_NAME.equals(mappingRule.getName()) && !mappingRule.getSubfields().isEmpty())
+      .flatMap(mappingRule -> mappingRule.getSubfields().get(0).getFields().stream())
+      .filter(mappingRule -> FUND_DISTRIBUTIONS_RULE_NAME.equals(mappingRule.getName()) && !mappingRule.getSubfields().isEmpty())
+      .flatMap(mappingRule -> mappingRule.getSubfields().get(0).getFields().stream())
+      .filter(mappingRule -> FUND_ID_RULE_NAME.equals(mappingRule.getName()))
+      .map(mappingRule -> ((Map<String, String>) mappingRule.getAcceptedValues()))
+      .findAny()
+      .orElse(Collections.emptyMap());
+  }
+
+  private void populateFundCode(org.folio.rest.jaxrs.model.FundDistribution fundDistribution, Map<String, String> idToFundName) {
+    String fundName = idToFundName.get(fundDistribution.getFundId());
+    fundDistribution.setCode(fundName.substring(fundName.lastIndexOf(LEFT_BRACKET) + 1, fundName.lastIndexOf(RIGHT_BRACKET)));
```

Comment:
can we get null here for fundName?

Reply:
we can not because this method is called when fundId is not null and idToFundName data is retrieved from mapping profile. fundId is populated when a mapping profile contains an appropriate pair of id and fund name that matches to fund name specified in the mapping for "fundId" field, so if "fundId" field is populated then a mapping profile and idToFundName map contains pair of fund id with corresponding fund name